app/components/EditProduct.js

```diff
@@ -0,0 +1,99 @@
+import React, { useEffect, useState } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { updateProduct } from '../redux/products';
+import TextField from '@material-ui/core/TextField';
+import Button from '@material-ui/core/Button';
+
+export default function EditProduct(props) {
+    const pageId = props.match.params.id;
+    const dispatch = useDispatch();
+
+    const user = useSelector((state) => state.user);
+    const products = useSelector((state) => state.products);
+
+    const [product, setProduct] = useState(null);
+    const [name, setName] = useState('');
+    const [description, setDescription] = useState('');
+    const [price, setPrice] = useState('');
+
+    useEffect(() => {
+        if (products.length && !product) {
+            const match = products.find(prod => prod.id == pageId);
+            setProduct(match);
+            setName(match.name);
+            setDescription(match.description);
+            setPrice(match.price);
+        }
+    });
+
+    const updateHandle = () => {
+        dispatch(updateProduct({ name, description, price, inventory: product.inventory }, product.id));
```

Comment:
Maybe should have option to update inventory too?

Reply:
Also - if name hasn't been entered in the form, won't name be entered as an empty string here? same for the other inputs. Maybe they should only be sent if they are truthy

Reply:
Earlier in the function, name is set to that of the product name once it matches (in the useEffect function). So if it isn't edited, the product name (or any other field) will still be the same.
I'll add a field for the inventory.